chain/jsonrpc/openapi/src/main.rs

```diff
@@ -0,0 +1,563 @@
+                    "application/json".to_string(),
+                    okapi::openapi3::MediaType {
+                        schema: Some(SchemaObject {
+                            reference: Some(response_schema_name),
+                            ..Default::default()
+                        }),
+                        ..Default::default()
+                    },
+                );
+                map
+            },
+            ..Default::default()
+        }
+        .into(),
+    );
+
+    let operation = okapi::openapi3::Operation {
+        operation_id: Some(method_name.clone()),
+        request_body: Some(request_body.into()),
+        responses,
+        ..Default::default()
+    };
+
+    let mut paths = PathsMap::new();
+    paths.insert(
+        format!("/{}", method_name),
+        okapi::openapi3::PathItem { post: Some(operation), ..Default::default() },
+    );
+
+    paths
+}
+
+fn add_spec_for_path_internal<RequestType: JsonSchema, ResponseType: JsonSchema>(
+    all_schemas: &mut SchemasMap,
+    all_paths: &mut PathsMap,
+    method_name: String,
+) {
+    let mut request_map = schemas_map::<RequestType>();
+    let response_map = schemas_map::<ResponseType>();
+
+    let request_struct_name = format!("JsonRpcRequest_for_{}", method_name);
+    let json_rpc_request = json!({
+        "properties": {
+            "id": {
+                "type": "string"
+            },
+            "jsonrpc": {
+                "type": "string"
+            },
+            "method": {
+                "enum": [
+                    method_name
+                ],
+                "type": "string"
+            },
+            "params": {
+                "$ref": format!("#/components/schemas/{}", RequestType::schema_name())
+            }
+        },
+        "required": [
+            "jsonrpc",
+            "id",
+            "params",
+            "method"
+        ],
+        "title": request_struct_name,
+        "type": "object"
+    });
+    if let Some(obj) = request_map.as_object_mut() {
+        obj.insert(request_struct_name.clone(), json_rpc_request);
+    }
+
+    let mut schemas = request_map;
+    okapi::merge::merge_map_json(&mut schemas, response_map, "name");
+
+    let paths = paths_map(
+        format!("#/components/schemas/{}", request_struct_name),
+        format!("#/components/schemas/{}", ResponseType::schema_name()),
+        method_name,
+    );
+
+    okapi::merge::merge_map_json(all_schemas, schemas.clone(), "name");
+    all_paths.extend(paths);
+}
+
+fn add_spec_for_path<Request: JsonSchema, Response: JsonSchema>(
+    all_schemas: &mut SchemasMap,
+    all_paths: &mut PathsMap,
+    method_name: String,
+) {
+    add_spec_for_path_internal::<Request, JsonRpcResponse<Response, RpcError>>(
+        all_schemas,
+        all_paths,
+        method_name,
+    )
+}
+
+fn whole_spec(all_schemas: SchemasMap, all_paths: PathsMap) -> OpenApi {
+    OpenApi {
+        openapi: "3.0.0".to_string(),
```

Comment:
Is it 3.0.0 or 3.1.0?

Reply:
3.0.0

Reply:
Actually it's because progenitor can't work with 3.1.0